core/src/main/java/io/micronaut/core/annotation/AnnotationValueBuilder.java

```diff
@@ -421,4 +421,17 @@
         }
         return this;
     }
+
+    /**
+     * Sets the members from the provided map
+     *
+     * @param members The map of members
+     * @return This builder
+     */
+    public AnnotationValueBuilder<T> members(Map<CharSequence, Object> members) {
+        if (members != null) {
+            values.putAll(members);
```

Comment:
Since this new API now allows arbitrary values I think we may need some checking that the objects in the map are either primitives, primitive arrays, strings, or annotation values otherwise passing something invalid could lead to odd compilation errors downstream